### PR TITLE
#5 Reactive skeleton

### PR DESCRIPTION
**Pull request by trenatov**
#5 Reactive skeleton

### CODE REVIEW

File: src/main/kotlin/io/openfuture/state/Application.kt [hunk 1]
```diff
@@ -9,4 +9,3 @@ class Application
 fun main(args: Array<String>) {
     runApplication<Application>(*args)
 }
-
```
Comment (ruslanys) [inline src/main/kotlin/io/openfuture/state/Application.kt:12]: Files should end with an empty line always https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline/729725

Comment (ruslanys) [reply src/main/kotlin/io/openfuture/state/Application.kt:12]: Check other files as well

File: src/test/kotlin/io/openfuture/state/base/MongoRepositoryTests.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import org.springframework.data.mongodb.core.convert.MongoConverter
+
+@DataMongoTest(excludeAutoConfiguration = [EmbeddedMongoAutoConfiguration::class])
+abstract class MongoRepositoryTests {
+
+    // @formatter:off
+    @Autowired
+    protected lateinit var mongoTemplate: MongoTemplate
```
Comment (ruslanys) [inline src/test/kotlin/io/openfuture/state/base/MongoRepositoryTests.kt]: at the latest versions of spring boot this bean is missing

So, just remove it

File: src/test/kotlin/io/openfuture/state/util/DummyData.kt [hunk 3]
```diff
@@ -0,0 +1 @@
+package io.openfuture.state.util
```
Comment (ruslanys) [inline src/test/kotlin/io/openfuture/state/util/DummyData.kt]: an empty line is missing

File: src/test/kotlin/io/openfuture/state/util/MockClient.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import com.nhaarman.mockitokotlin2.given
+import com.nhaarman.mockitokotlin2.mock
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.core.io.ClassPathResource
+import org.springframework.http.HttpHeaders
+import org.springframework.web.reactive.function.client.ClientResponse
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+
```
Comment (ruslanys) [inline src/test/kotlin/io/openfuture/state/util/MockClient.kt]: This is a very specific class. Don't think we will need it. So, remove it.

**Review by ruslanys — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/openfuture/state/Application.kt (hunk 1 above)
Comment (trenatov) [reply src/main/kotlin/io/openfuture/state/Application.kt:12]: This file had two empty lines, so i removed one

**Comment by trenatov** [comment]
@ruslanys made changes

**Review by ruslanys — APPROVED** [review]

**Pull request merged**